src/fondant/pipeline/runner.py

```diff
@@ -92,11 +97,37 @@ def _resolve_imports(self):
             )
 
     def run(
```

Comment:
Can't help but notice that the flow of this method is the same for every runner. Might be worth it to lift it to the base `Runner` class. Up to you.

Reply:
I had the same idea when commuting home, will also make testing a lot easier